### ISSUE TITLE
Create a separate package for the icons

### ISSUE THREAD

**Issue by tordans** [enzet/map-machine#149]
The id-tagging-schema project started using the Röntgen icons and with it several OSM editor projects now have pull the icons and package those that are used in presets as part of their apps.

My understanding is, that it is not easy to import just the icon set. In https://github.com/ideditor/schema-builder/issues/75#issuecomment-1410038964 the request was made to have a separate package for just the icons.
> //cc @enzet: It would be superbly nice to have a separate versioned package of just the pure icons, in order to use them more easily outside of map-machine. Do you think something like that would be feasible?

I created this issue to give this more prominence and track it more easily.

For more details on the technical side @tyrasd, @bryceco, @westnordost and other id-tagging-schema-consumers can give more insights.

**Comment by matkoniecz** [comment]
I tried to use one of icons in iD preset but I am unable to guess its identifier 

![screen03](https://user-images.githubusercontent.com/899988/229084579-8ba1c18d-d200-4ee3-85c1-9e3181583946.png)

From the second row.

**Comment by tyrasd** [comment]
it's the `fort` (https://github.com/openstreetmap/iD/blob/develop/svg/roentgen/fort.svg)

**Comment by enzet** [comment]
Sorry guys, I've been out of the project for a while.

@tordans, thank you for creating the issue. I'm glad that the icon set is living his live now outside of the Map Machine project.

@tyrasd, thank you for writing a readme about the icons and for deliberately placing the license. I was thinking about creating a separate repository for Röntgen icons, however, I switched to the new project, [iconscript](https://github.com/enzet/iconscript), which makes the icon set even more generated. So, my final plan was to put everything in three separate repositories:
  - __Map Machine__ for the map renderer and mapping scheme,
  - __Röntgen__ for SVG templates and iconscript files,
  - __iconscript__ for the icon renderer.

As for the icons, it was planned to create a separate web page and make them downloadable through web interface as SVG files.

Is this plan suits you or do you want the final SVG files to be in the GitHub repository in order to easily track commits or something like that?

**Comment by tordans** [comment]
> Is this plan suits you or do you want the final SVG files to be in the GitHub repository in order to easily track commits or something like that?

Generally speaking, a good model for iD/Rapid/GoMap etc. would be to have a Github repo for the icons that follows the structure that https://github.com/rapideditor/temaki uses, especially:

- The icons are SVGs in the repo and viewable and searchable (https://github.com/rapideditor/temaki/tree/main/icons)
- The repo uses semantic versioning with a package json 
- The repos release as npm package like https://npm.io/package/@rapideditor/temaki so we can easily include it in other (JS) projects
- Optionally: There is a page to view the icons to make it even easier to pick one like https://rapideditor.github.io/temaki/docs/ – I find this really optional, since IMO at some point we should create a shared page for the tagging-schema that makes it easier to pick icons from all the available sources.

Alternatively, you could look into contributing the icons to temaki and we all use this one place. However, I would start by getting input by the maintainers if some of the icons are not a good fit for temaki; if so, having something separate is easer overall.

**Comment by westnordost** [comment]
Wow, looks pretty complex distribution for just some icons.

For StreetComplete (which also uses iD icons), most convenient would be if there was a publicly accessible directory somewhere from which they could be downloaded by name.

**Comment by enzet** [comment]
@tordans Sounds good!

  - A separate repo with `icons` directory—no problem.
  - I'll definitely use semantic versioning, however I want to mix it with versions for individual icons, so that `icons` directory may contain, say, `ford_v1.svg` and `ford_v2.svg` icons (or something like that).
  - I know almost nothing about npm so I have to research this part. If you can provide me with links to some useful information, this will be great.
  - Web page is definitely on the roadmap. I even have [demo version](http://enzet.ru/roentgen/) but it is outdated and doesn't have download button.

Contributing to Temaki doesn't seem to me like good idea, because the overall style of Röntgen is slightly different and the process of creating and changing icons is more complex.

@westnordost I definitely overengineered it. 🥲

Yes, there will be a web site with icons as well as a directory in a GitHub repository.

**Comment by tordans** [comment]
> I know almost nothing about npm …

Me neither but I am told for simple use cases it is straight forward. @tyrasd can maybe give a few hints…

**Comment by tordans** [comment]
> …however I want to mix it with versions for individual icons, so that icons directory may contain, say, ford_v1.svg and ford_v2.svg icons (or something like that).

Personally I would be happy to alway get the one icon from the current package. Instead of using v1,v2 maybe just add a new icon instead. So there could be "tower.svg" and "tower-windows.svg" for a version that shows windows as well…

**Comment by enzet** [comment]
@tordans Yeah, I thought about it and it seems like semver should deal with this problem. Do I understand it right, that if I want to drastically change the shape of the icon, I should just bump the major version of the whole package and this will be OK?

For example, is it OK if those icons have the same name `diving_platform.svg`, but come from versions 1.0, 2.0, 2.1 respectivelly?

![diving_platform_icons](https://github.com/enzet/map-machine/assets/2399987/09efc1ea-beaa-4cde-9643-4686c8bac8be)

**Comment by enzet** [comment]
It took me an embarrassing amount of time, but I’ve finally separated the Map Machine (renderer) and Röntgen (icon set) projects. Röntgen now uses semantic versioning (current version: 0.12.0) and contains 514 icons.

To use the icons, one can:
- search, browse, and download them from the [Röntgen website](https://enzet.ru/roentgen),
- install the npm package [@enzet/roentgen](https://www.npmjs.com/package/@enzet/roentgen) with `npm install @enzet/roentgen`,
- install the PyPI package [roentgen-icons](https://pypi.org/project/roentgen-icons/) with `pip install roentgen-icons`,
- download the latest development version from the [`icons`](icons) directory,
- download a ZIP file for a specific release from [GitHub Releases](https://github.com/enzet/Roentgen/releases).

@tordans Since this issue was opened, I’ve redrawn many icons while keeping their identifiers unchanged. I’d be happy to update the icons in schema-builder and Rapid. Should I open PRs with the updated icons, or would you prefer switching to the npm package?

@westnordost Should I also create a Kotlin package (or something similar) for StreetComplete?

**Comment by enzet** [comment]
> How would you do it, by the way?

I was also planning to write a script. I already have some scripts for parsing SVG path commands, so it doesn’t seem too hard.

**Comment by enzet** [comment]
Thank you so much for quick response!
